### ISSUE TITLE
WARNING: failed to decrypt with DPAPI-----ERROR: 'NoneType' object has no attribute 'decode'

### ISSUE THREAD

**Issue by bigestyjoy** [yt-dlp/yt-dlp#10927]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

Happens on all sites.
After removing the "--cookies-from-browser chrome", the error does not occur.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU -f 248+251 --proxy http://127.0.0.1:10809 --cookies-from-browser chrome --mark-watched https://www.youtube.com/watch?v=v9lMa4BGAuA
[debug] Command-line config: ['-vU', '-f', '248+251', '--proxy', 'http://127.0.0.1:10809', '--cookies-from-browser', 'chrome', '--mark-watched', 'https://www.youtube.com/watch?v=v9lMa4BGAuA']
[debug] Encodings: locale cp936, fs utf-8, pref cp936, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.06 from yt-dlp/yt-dlp [4d9231208] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.26120-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2023-07-19-git-efa6cec759-full_build-www.gyan.dev (setts), ffprobe 2023-07-19-git-efa6cec759-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {'all': 'http://127.0.0.1:10809'}
Extracting cookies from chrome
[Cookies] Searching for "Cookies":      1 files searched
[debug] Extracting cookies from: "C:\Users\biges\AppData\Local\Google\Chrome\User Data\Default\Network\Cookies"
[debug] Found local state file at "C:\Users\biges\AppData\Local\Google\Chrome\User Data\Local State"
[Cookies] Loading cookie   1842/  2088WARNING: failed to decrypt with DPAPI
Traceback (most recent call last):
  File "yt_dlp\__main__.py", line 17, in <module>
  File "yt_dlp\__init__.py", line 1081, in main
  File "yt_dlp\__init__.py", line 979, in _real_main
  File "yt_dlp\YoutubeDL.py", line 720, in __init__
  File "yt_dlp\YoutubeDL.py", line 4070, in print_debug_header
  File "functools.py", line 967, in __get__
  File "yt_dlp\YoutubeDL.py", line 4245, in _request_director
  File "yt_dlp\YoutubeDL.py", line 4220, in build_request_director
  File "functools.py", line 967, in __get__
  File "yt_dlp\YoutubeDL.py", line 4116, in cookiejar
  File "yt_dlp\cookies.py", line 94, in load_cookies
  File "yt_dlp\cookies.py", line 115, in extract_cookies_from_browser
  File "yt_dlp\cookies.py", line 315, in _extract_chrome_cookies
  File "yt_dlp\cookies.py", line 350, in _process_chrome_cookie
  File "yt_dlp\cookies.py", line 525, in decrypt
AttributeError: 'NoneType' object has no attribute 'decode'
[17148] Failed to execute script '__main__' due to unhandled exception!
```

**Comment by seproDev** [comment]
Since there are multiple reports, this seems to be a change in Chrome. I cannot replicate on my machine.

This could potentially be related to https://security.googleblog.com/2024/07/improving-security-of-chrome-cookies-on.html
which ones rolled out will likely mean `--cookies-from-browser chrome` will be permanently broken on Windows.

**Comment by Snickbrack** [comment]
Can we bypass this with using VirtualBox and Ubuntu for example?

Will the cookie-Option be available there again?

**Comment by seproDev** [comment]
Possible workarounds are:
- Use Firefox
- Use Linux
- Export the cookies with an extension such as [this](https://chromewebstore.google.com/detail/get-cookiestxt-locally/cclelndahbckbenkjhflpdbgdldlbecc) and use `--cookies` instead
- Use the `ApplicationBoundEncryptionEnabled` enterprise rule as outlined [here](https://chromeenterprise.google/policies/#ApplicationBoundEncryptionEnabled)

**Comment by yynopenguin** [comment]
Same problem. Used seproDev's local cookies method to solve

**Comment by pyt00n00b** [comment]
Setting "ApplicationBoundEncryptionEnabled" enterprise rule to "Disabled" through group policy:

![ApplicationBoundEncryptionEnabled_02](https://github.com/user-attachments/assets/7897c16b-6498-48bf-a035-3ed33eda4b90)

 or through registry:

```
[HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome]
"ApplicationBoundEncryptionEnabled"=dword:00000000
```

```
[HKEY_CURRENT_USER\Software\Policies\Google\Chrome]
"ApplicationBoundEncryptionEnabled"=dword:00000000
```

 has **NO** effect on the DPAPI error.

**Comment by seproDev** [comment]
Hmm, in my testing before posting it seemed to work with the policy showing up in `chrome://policy`, and me being able to decrypt cookies.
Though now, no matter if I set it to true or false, I cannot replicate the error 🤔

**Comment by Snickbrack** [comment]
Okay I have just exported the cookies from Chrome with the Plugin mentioned above and works fine.

I even dont need to close Chrome now to be able to use the Cookies in yt-dlp. Nice!

**Comment by bigestyjoy** [comment]
> Possible workarounds are:
> 
> * Use Firefox
> * Use Linux
> * Export the cookies with an extension such as [this](https://chromewebstore.google.com/detail/get-cookiestxt-locally/cclelndahbckbenkjhflpdbgdldlbecc) and use `--cookies` instead
> * Use the `ApplicationBoundEncryptionEnabled` enterprise rule as outlined [here](https://chromeenterprise.google/policies/#ApplicationBoundEncryptionEnabled) (maybe)

"Export the cookies with an extension such as [this](https://chromewebstore.google.com/detail/get-cookiestxt-locally/cclelndahbckbenkjhflpdbgdldlbecc) and use --cookies instead"

This works.
Export the cookies to a local file.
Replace "--cookies-from-browser chrome" By "-cookies ...\cookies.txt"

**Comment by jim60105** [comment]
Notice that you may need to export the cookie in a private/incognito window because of https://github.com/yt-dlp/yt-dlp/issues/8227#issuecomment-1867184804

**Comment by Remzi1993** [comment]
I also have this error after using --cookies-from-browser chrome

**Comment by raki-can** [comment]
> Possible workarounds are:
> * Export the cookies with an extension such as [this](https://chromewebstore.google.com/detail/get-cookiestxt-locally/cclelndahbckbenkjhflpdbgdldlbecc) and use `--cookies` instead

For me this worked out well.

**Comment by awei78** [comment]
> Setting "ApplicationBoundEncryptionEnabled" enterprise rule to "Disabled" through group policy:
> 
> ![ApplicationBoundEncryptionEnabled_02](https://private-user-images.githubusercontent.com/180181643/365084670-7897c16b-6498-48bf-a035-3ed33eda4b90.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NzczODIsIm5iZiI6MTcyNTg3NzA4MiwicGF0aCI6Ii8xODAxODE2NDMvMzY1MDg0NjcwLTc4OTdjMTZiLTY0OTgtNDhiZi1hMDM1LTNlZDMzZWRhNGI5MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwOVQxMDE4MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZmM2ZWFmMmQzM2ZhOWZkY2MzMDYyMGQyNjY0YzQzMWU1OGY0YzM4ZTljZjY5YmM4MzFlODQyOWU1YzAwNGViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jeC4zDEX3adDb5dXLRTq34dJ92Qp3NL_7Aoz22ARxWI)
> 
> or through registry:
> 
> ```
> [HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome]
> "ApplicationBoundEncryptionEnabled"=dword:00000000
> ```
> 
> ```
> [HKEY_CURRENT_USER\Software\Policies\Google\Chrome]
> "ApplicationBoundEncryptionEnabled"=dword:00000000
> ```
> 
> has **NO** effect on the DPAPI error.

Yes, it has no effect

**Comment by seproDev** [comment]
Okay, what I assume is going on is that you are testing with your already encrypted user data, while I was clearing my user data.
```
  // Policy allows disabling App-Bound encryption. Note, this will not disable
  // decryption of existing data.
```
https://github.com/chromium/chromium/blob/2729d053c0076d989db58e63a48f168e8938dc10/chrome/browser/os_crypt/app_bound_encryption_win.cc#L40-L41

Reading that, I would assume that once your data is encrypted, there is no way to revert that change. There may be other things going on, but I don't have the time to further investigate right now.

**Comment by Fannovel16** [comment]
I tried to comment part https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/cookies.py#L1055-L1057 and for some reasons, it works

**Comment by fireattack** [comment]
[rookie](https://github.com/thewh1teagle/rookie) and its py binding [rookiepy](https://github.com/thewh1teagle/rookie/tree/main/bindings/python) can still get cookies from Chrome after this update. Maybe we can follow their method?

**Comment by gamer191** [comment]
@fireattack I haven't checked if that fixes this issue, but it definitely fixes #7271 (even without admin). Notably, it has the downside that it only gets cookies that were created before the current session.

Perhaps they found a location where Chrome backs up it's cookies🤔

**Comment by gamer191** [comment]
> [rookie](https://github.com/thewh1teagle/rookie) and its py binding [rookiepy](https://github.com/thewh1teagle/rookie/tree/main/bindings/python) can still get cookies from Chrome after this update

It didn't work for me, see https://github.com/thewh1teagle/rookie/issues/50

**Comment by fireattack** [comment]
> > [rookie](https://github.com/thewh1teagle/rookie) and its py binding [rookiepy](https://github.com/thewh1teagle/rookie/tree/main/bindings/python) can still get cookies from Chrome after this update
> 
> It didn't work for me, see [thewh1teagle/rookie#50](https://github.com/thewh1teagle/rookie/issues/50)

Yeah my bad, I should be more clear. 

When I was testing, I used it with specific domains, and since presumably these cookies are still using older encryption methods (v10, v11, etc.), it worked. 

Currently, yt-dlp's importing will break when *any* cookie is v20. This might be a overkill for the user. `gallery-dl` has a nice syntax of `--cookies-from-browser chrome/nicovideo.jp` to only import cookies for certain site. Maybe we can implement something similar so we don't need to import all the cookies all the time and thus avoiding reading v20 cookies.

Not sure which cookies would be v20, though: is it site-specific, or depending one when the cookie was created?

**Comment by y4my4my4m** [comment]
I'd rather not have my cookies as plaintext on my computer though, any other solution?

**Comment by gamer191** [comment]
> Not sure which cookies would be v20, though: is it site-specific, or depending one when the cookie was created?

I think it depends which Chrome version created the cookies
Hence having that feature would be somewhat useful, but only for a very short time period
> I'd rather not have my cookies as plaintext on my computer though, any other solution?

I don't think --cookies-from-browser ever saves the cookies, although obviously it would be a different situation if it was --cookies

Depending on the site, you might be able to use `--username USERNAME` instead. It will then interactively prompt you for the password, preventing the password from being included in any shell logs

Note: whilst I do definitely see the security benefit to Chrome's new approach, I think it's important to note that cookies will be a small concern if you get malware on your computer (even if it doesn't have admin permissions). For a start, websites you haven't visited since updating Chrome will still have unencrypted cookies

**Comment by ricnava00** [comment]
> Hmm, in my testing before posting it seemed to work with the policy showing up in `chrome://policy`, and me being able to decrypt cookies. Though now, no matter if I set it to true or false, I cannot replicate the error 🤔

If I understand correctly, this is a bug and it has been fixed in https://github.com/chromium/chromium/commit/5c5e9f87a6c742fa5a137cdcbeeb13b4d9531cca
Unfortunately this is planned for chrome 130 so it will take some time until it's available

**Comment by EntityinArray** [comment]
`'NoneType' object has no attribute 'decode'`
instead of waiting for chrome to update, we can not call `decode` on a `NoneType`

**Comment by Grub4K** [comment]
> instead of waiting for chrome to update, we could not call `decode` on a `NoneType`

...which would be equivalent to not passing `--cookies-from-browser chrome` since all the cookies still fail to decrypt and therefore is not a viable solution

**Comment by mic-user** [comment]
Guys, 
come to https://github.com/borisbabic/browser_cookie3/issues/210
and please gather/show some ideas for the solution.

**Comment by mic-user** [comment]
A workaround posted at
https://github.com/borisbabic/browser_cookie3/issues/210#issuecomment-2352627713.

Have a good day.

**Comment by klikevil** [comment]
> A workaround posted at [borisbabic/browser_cookie3#210 (comment)](https://github.com/borisbabic/browser_cookie3/issues/210#issuecomment-2352627713).
> 
> Have a good day.

As posted before in this, 

[HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome]
"ApplicationBoundEncryptionEnabled"=dword:00000000

DOES NOT solve this issue.



> > Setting "ApplicationBoundEncryptionEnabled" enterprise rule to "Disabled" through group policy:
> > ![ApplicationBoundEncryptionEnabled_02](https://private-user-images.githubusercontent.com/180181643/365084670-7897c16b-6498-48bf-a035-3ed33eda4b90.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NzczODIsIm5iZiI6MTcyNTg3NzA4MiwicGF0aCI6Ii8xODAxODE2NDMvMzY1MDg0NjcwLTc4OTdjMTZiLTY0OTgtNDhiZi1hMDM1LTNlZDMzZWRhNGI5MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwOVQxMDE4MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZmM2ZWFmMmQzM2ZhOWZkY2MzMDYyMGQyNjY0YzQzMWU1OGY0YzM4ZTljZjY5YmM4MzFlODQyOWU1YzAwNGViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jeC4zDEX3adDb5dXLRTq34dJ92Qp3NL_7Aoz22ARxWI)
> > or through registry:
> > ```
> > [HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome]
> > "ApplicationBoundEncryptionEnabled"=dword:00000000
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > [HKEY_CURRENT_USER\Software\Policies\Google\Chrome]
> > "ApplicationBoundEncryptionEnabled"=dword:00000000
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > has **NO** effect on the DPAPI error.
> 
> Yes, it has no effect

**Comment by klikevil** [comment]
> > A workaround posted at [borisbabic/browser_cookie3#210 (comment)](https://github.com/borisbabic/browser_cookie3/issues/210#issuecomment-2352627713).
> > Have a good day.
> 
> As posted before in this,
> 
> [HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome] "ApplicationBoundEncryptionEnabled"=dword:00000000
> 
> DOES NOT solve this issue.
> 
> > > Setting "ApplicationBoundEncryptionEnabled" enterprise rule to "Disabled" through group policy:
> > > ![ApplicationBoundEncryptionEnabled_02](https://private-user-images.githubusercontent.com/180181643/365084670-7897c16b-6498-48bf-a035-3ed33eda4b90.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NzczODIsIm5iZiI6MTcyNTg3NzA4MiwicGF0aCI6Ii8xODAxODE2NDMvMzY1MDg0NjcwLTc4OTdjMTZiLTY0OTgtNDhiZi1hMDM1LTNlZDMzZWRhNGI5MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwOVQxMDE4MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZmM2ZWFmMmQzM2ZhOWZkY2MzMDYyMGQyNjY0YzQzMWU1OGY0YzM4ZTljZjY5YmM4MzFlODQyOWU1YzAwNGViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jeC4zDEX3adDb5dXLRTq34dJ92Qp3NL_7Aoz22ARxWI)
> > > or through registry:
> > > ```
> > > [HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome]
> > > "ApplicationBoundEncryptionEnabled"=dword:00000000
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > ```
> > > [HKEY_CURRENT_USER\Software\Policies\Google\Chrome]
> > > "ApplicationBoundEncryptionEnabled"=dword:00000000
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > has **NO** effect on the DPAPI error.
> > 
> > 
> > Yes, it has no effect

The only solution I have found involves enabling remote debugger on chrome in its shortcut and querying the json from that, so if we want this back to windows someone would need to make a pull that modifies the chrome shortcut doing that and querying from the remote debugger.

**Comment by mic-user** [comment]
@klikevil 
As I post multiple steps, the entire full steps works with the registry.
Since the registry is NOT automatic switch, but only a trigger for following steps.
Kid, you must DO the entire steps before you announc rejection that shame.
Plus, 'HKEY_CURRENT_USER' section does not effect to the switch. Chrome only recognize HKEY_LOCAL_MACHINE section.
So,
please keep silence if you dont know entire mechanism kid.

**Comment by brentil** [comment]
Has anyone tried clearing cookies after setting the ApplicationBoundEncryptionEnabled and rebuilding them?  As mentioned above it doesn't decrypt once encrypted but it might prevent the encryption of new ones.

**Comment by rgiuffre** [comment]
I am getting the same error using Microsoft Edge, and do not, nor have I ever installed Chrome on this computer. I did, however, install a Chrome extension recently into Edge. I don't know if that makes a difference or not. Please advise.
Thanks.

**Comment by seproDev** [comment]
@rgiuffre Edge uses Chromium under the hood. All Chromium based browsers are affected.

**Comment by ricnava00** [comment]
Looking at https://github.com/chromium/chromium/blob/99fdd0d45b56eddea940b62796c297b2334e0dc9/chrome/elevation_service/elevator.cc#L198, it seems that Chrome has some more processing after the standard DPAPI decryption

This doesn't seem to be the case for Edge, and probably other Chromium-based browsers

The new encryption method is basically the same as before with one extra step

Before, we had to take encrypted_key, and decrypt it with the user's DPAPI. This gave an AES key to decrypt the cookies

Now, app_bound_encrypted_key just contains a different "encrypted_key"-like key, encrypted a second time
You just need to decrypt app_bound_encrypted_key with the SYSTEM DPAPI, and you will get another key which works exactly as encrypted_key did, meaning that you can decrypt this key again with the user's DPAPI and use the result to AES-decrypt the cookies

A quick way to check if this works is to decrypt the app_bound_encrypted_key with SYSTEM's DPAPI, decrypt the result with the user's DPAPI, and if the result is 32 bytes long, that's a valid AES key

With Edge, I get a valid 32 byte key
With Chrome, the key I obtain is 61 bytes, since as I said before Chrome does another step which has no public source AFAIK

BTW, if you hate yourself, you can debug chrome.dll with something like WinDBG and their symbol server (https://chromium-browser-symsrv.commondatastorage.googleapis.com) and print the AES key after it gets decrypted

**Comment by blul1ghtz** [comment]
if anybody has information on the new v20 encryption being rolled out lmk cus i got questions like is it the same decrytion method and by application bound does it mean the application pid or the application name

**Comment by klikevil** [comment]
> if anybody has information on the new v20 encryption being rolled out lmk cus i got questions like by application bound does it mean the application pid or the application name

LOL @blul1ghtz has one public repo and it's a "chromium infostealer", the very people they're trying to prevent from getting access to these in the first place 🤣.  Sorry I know it's not exactly on the bug topic, but had to point out the irony.

**Comment by blul1ghtz** [comment]
hey i jsut work on projects i want to and this year im working on sketchier projects is all, never intended for somebody to act up like this over something soo small

**Comment by runassu** [comment]
Chrome cookie encrypted_value v20 use `app_bound_encrypted_key` in Local State file. To decrypt this, we first need to decrypt `app_bound_encrypted_key` with the SYSTEM DPAPI, followed by the user DPAPI. In other brand browsers, we can directly get the 32-bytes AES key to decrypt encrypted cookies. Chrome requires some additional steps.
ref:
[https://github.com/chromium/chromium/blob/35afbc6f6b81d51d697ea615364a972832dab418/chrome/elevation_service/elevator.cc#L199](url)

For example, after the double step DPAPI decryption, the resulting value comes with Chrome path, then 1-byte flag 0x01, 12-bytes IV, 32-bytes ciphertext, 16-bytes TAG.
```
00000000  1f 00 00 00 02 43 3a 5c 50 72 6f 67 72 61 6d 20  |.....C:\Program |
00000010  46 69 6c 65 73 5c 47 6f 6f 67 6c 65 5c 43 68 72  |Files\Google\Chr|
00000020  6f 6d 65 3d 00 00 00 01 ca bf 17 e5 f2 f4 47 b0  |ome=....Ê¿.åòôG°|
00000030  e8 1b 64 1b f2 7c 22 49 66 e2 5f fc ed d2 e0 cf  |è.d.ò|"Ifâ_üíÒàÏ|
00000040  c0 4e 1f 21 f6 1b c2 da a2 eb 6f 53 2c 47 d3 9e  |ÀN.!ö.ÂÚ¢ëoS,GÓ.|
00000050  7b 50 e6 7f 4d 5c 34 3f e6 ee d9 43 58 91 9e d2  |{Pæ.M\4?æîÙCX..Ò|
00000060  3a d8 96 30                                      |:Ø.0|
```
IV: `ca bf 17 e5 f2 f4 47 b0 e8 1b 64 1b`
ciphertext: `f2 7c 22 49 66 e2 5f fc ed d2 e0 cf c0 4e 1f 21 f6 1b c2 da a2 eb 6f 53 2c 47 d3 9e 7b 50 e6 7f `
TAG: `4d 5c 34 3f e6 ee d9 43 58 91 9e d2 3a d8 96 30`
We can decrypt it using AES-256-GCM, with the key hardcoded in `elevation_service.exe`:
```
01455184   B3 1C 6E 24 1A C8 46 72  8D A9 C1 FA C4 93 66 51
01455200   CF FB 94 4D 14 3A B8 16  27 6B CC 6D A0 28 47 87
```
This will yield the decrypted key, which works exactly as `encrypted_key` did.
```
00000000  6d 29 6e e5 7a 29 25 6e 74 5e 26 25 15 97 1e 66
00000010  c1 98 cd 32 2c a6 9f fd 57 de 15 73 8b ed cd 6c
```
Here is th PoC
[https://gist.github.com/runassu/216159c55b5fc895f14dd5f577a72e85]()

**Comment by blul1ghtz** [comment]
omg thank you soo much i will look into this while i got time, looks very promising.

**Comment by mic-user** [comment]
@runassu Nice.
One question to you that have you tested on the new chrome version 130.0.6723.59 which released on 16th Oct 2024?
The reason whay asking you is that they made small format changed for encrypted data even for v10(for example to the cookie, they put some extra 32bytes data at the begining of the decrypted stream.).
So that if you possible to check, you might verify the final v20 app bounded decrypted key is okay as is or need to make some changes for 130.0.6723.59 or later for the people.
Thanks.

**Comment by runassu** [comment]
> @runassu Nice. One question to you that have you tested on the new chrome version 130.0.6723.59 which released on 16th Oct 2024? The reason why asking you is that the google made small format changed for encrypted data even for v10(for example to the cookie, they put some extra 32bytes data at the begining of the decrypted stream.). So that if you possible to check, you might verify the final v20 app bounded decrypted key is okay as is or need to make some changes for 130.0.6723.59 or later for the people.
> 
> [what i am trying to say is]
> 
> 1. from 'key = cipher.decrypt_and_verify(ciphertext, tag)' to key = cipher.decrypt_and_verify(ciphertext, tag)[32:]
> 2. decrypted_cookie = cookie_cipher.decrypt_and_verify(encrypted_cookie, cookie_tag).decode()  >> decrypted_cookie = cookie_cipher.decrypt_and_verify(encrypted_cookie, cookie_tag)[32:].decode()
> 
> Thanks.

@mic-user the 32-bytes extra data is the SHA256 hash of the domain value, nothing to do with the decryption algo
ref:
[https://source.chromium.org/chromium/chromium/src/+/main:net/extras/sqlite/sqlite_persistent_cookie_store.cc;l=223](https://source.chromium.org/chromium/chromium/src/+/main:net/extras/sqlite/sqlite_persistent_cookie_store.cc;l=223)

**Comment by mic-user** [comment]
@runassu Thanks. God bless you.

**Comment by ricnava00** [comment]
Since chrome 130 is out, setting
[HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome] "ApplicationBoundEncryptionEnabled"=dword:00000000
should now work
I'm not sure if v20 cookies are automatically downgraded to v10, but if not, they will probably be saved as v10 when their value changes or they can be manually deleted and recreated.

We have seen that you need admin privileges to decrypt the app-bound key, and the flow would probably take time to be implemented

I'd say that first step could be fixing the v10 handling for users that set the registry key, so
1. Ignore v20 cookies
2. Skip the first 32 bytes as discussed in the previous comments (possibly enabling this behavior only if meta.version is >=24)

**Comment by xyzxye** [comment]
I don't know if it's just me who has encountered this problem. When I use the Chrome extensions like [Get cookies.txt LOCALLY](https://chromewebstore.google.com/detail/get-cookiestxt-locally/cclelndahbckbenkjhflpdbgdldlbecc), I find that cookies are changing? For example, the cookies file from an hour ago became invalid :(

like `SIDCC`, `__Secure-3PSIDTS` and so on

**Comment by bashonly** [comment]
@xyzxye https://github.com/yt-dlp/yt-dlp/wiki/Extractors#exporting-youtube-cookies

**Comment by RileyXX** [comment]
I used  `--cookies-from-browser firefox` instead. Worked like a charm. Didn't have to close the browser either.

**Comment by blul1ghtz** [comment]
^^ true, lets just hope firefox doesnt change there encryption standards

**Comment by thewh1teagle** [comment]
FYI: Added support for the latest Chrome encryption in [rookie](https://github.com/thewh1teagle/rookie). Now compatible with Python, JavaScript, and Rust

**Comment by tailinhcapricorn13th** [comment]
[runassu](https://github.com/runassu) 
how can you get this code aes_key = binascii.a2b_base64("sxxuJBrIRnKNqcH6xJNmUc/7lE0UOrgWJ2vMbaAoR4c=")

**Comment by mimi89999** [comment]
There is also w way to extract cookies from Chrome that doesn't require admin rights 😉 

See https://github.com/thewh1teagle/rookie/issues/81#issuecomment-2453255147

**Comment by mic-user** [comment]
@yidi000 
Add the registry
[HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome]
"ApplicationBoundEncryptionEnabled"=dword:00000000
And reboot or kill Chrome process all manually(visible and invisiable processes)
That is works of cause since Google official unless not following the instruction. Ignore the wrong information.

**Comment by mic-user** [comment]
To all,
who want to use v10 decryption method,
follow the steps at 
1. https://github.com/borisbabic/browser_cookie3/issues/210#issuecomment-2352627713.

If you going to use v20, 
follow the script that 
1. https://github.com/yt-dlp/yt-dlp/issues/10927#issuecomment-2412854548 
or
2. https://gist.github.com/thewh1teagle/359675c2f5ea4920949448ec705f9fb2

**Comment by yidi000** [comment]
> @yidi000 Add the registry [HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome] "ApplicationBoundEncryptionEnabled"=dword:00000000 And reboot or kill Chrome process all manually(visible and invisiable processes) That is works of cause since Google official unless not following the instruction. Ignore the wrong information.

sorry. where should I add the registry? thx

**Comment by mic-user** [comment]
@yidi000 
At 'HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome' as the registry pointing out.

**Comment by pro-arch-user** [comment]
> Possible workarounds are:
> 
> * Use Firefox
> * Use Linux
> * Export the cookies with an extension such as [this](https://chromewebstore.google.com/detail/get-cookiestxt-locally/cclelndahbckbenkjhflpdbgdldlbecc) and use `--cookies` instead
> * Use the `ApplicationBoundEncryptionEnabled` enterprise rule as outlined [here](https://chromeenterprise.google/policies/#ApplicationBoundEncryptionEnabled) (maybe)

--cookies worked for me!

**Comment by blul1ghtz** [comment]
another option is using the devtools protocol (path/to/chrome.exe --remote-debugging-port=PORT_NUM) and send a payload with network.getCookies (not sure exactly how the payload is layed out but it sends back raw cookie arrays for whatever sites you specify), more info at https://chromedevtools.github.io/devtools-protocol/tot/Network/

**Comment by TrulyStucker** [comment]
Hey everyone, can anyone help, I think I did everything according to advises below, I extracted cookies using get cookies.txt locally and still stumble across 403 error (access forbidden). Trying to DL vid from Patreon on Windows.

`yt-dlp --cookies "D:\ffmpeg\bin\cookies.txt" --referer "https://www.patreon.com/posts/wintersun-sons-110460144" https://vod-adaptive-ak.vimeocdn.com/exp=1733331699~acl=%2F17615183-bd54-4774-ae2f-5e2f01278e32%2F%2A~hmac=10f82d18aa94cbb2a9b56bad2b16058891de49acfc64e34f6c709116b03d0453/17615183-bd54-4774-ae2f-5e2f01278e32/v2/playlist/av/primary/prot/cXNyPTE/playlist.m3u8c`

Basically this is my command, browser is closed.

Here is what I'm getting:

[generic] Extracting URL: https://vod-adaptive-ak.vimeocdn.com/exp=1733331699~acl=%2F17615183-bd54-4774-ae2f-5e2f01278e32%2...t.m3u8?omit=av1-hevc
[generic] playlist: Downloading webpage
ERROR: [generic] Unable to download webpage: HTTP Error 403: Forbidden (caused by <HTTPError 403: Forbidden>)

Any help is appreciated.

**Comment by TrulyStucker** [comment]
> @TrulyStucker this is the wrong place. This issue is specifically for tracking `--cookies-from-browser` not working. With that said just passing the patreon post URL should work. If you need further help, please open a new issue and provide a full verbose log.

Yes, '--cookies-from-browser' is not working, that's why I came her for a workaround hoping someone will check if my syntax and code for that workaround works. Because 403 error is returned due to yt-dlp was unable to parse cookies from file (as far as I think). So I wouldn't call it off-topic.

**Comment by seproDev** [comment]
@TrulyStucker No, this is off topic. Site specific questions where you are not even using `--cookies-from-browser` are not related to this issue. I mentioned a possible solution. If it doesn't work open a new issue and provide a complete verbose log. Without a complete verbose log we can not help you.

**Comment by Rootkit-** [comment]
Work around... Install opera. Install "EXP Cookies Tool" on Chrome/Brave and opera. Export cookies and import into opera. I know this isn't solving the issue but for many it will create workaround.
yt-dlp --cookies-from-browser opera --cookies cookie.txt

**Comment by vlakoff** [comment]
Another workaround. Personally I'm fine with just reading cookies from my main Firefox profile, but if desired, you may create a dedicated profile:

1. Start Firefox with a new profile, in the specified `<profile dir>`:

```
"C:\Program Files\Mozilla Firefox\firefox.exe" --no-remote --profile "<profile dir>"
```

* The first call of the command will automatically create the profile.
* The `--no-remote` parameter lets you run it concurrently with other Firefox profiles.

2. On that Firefox instance, log into YouTube, etc. You may then close the Firefox instance.

3. Use yt-dlp, reading from these cookies:

```
yt-dlp --cookies-from-browser firefox:"<profile dir>" https://www.youtube.com/watch?v=dQw4w9WgXcQ
```

**Comment by RichardWinters** [comment]
> Work around...
> 
> 1. Install opera.
> 2. Install "EXP Cookies Tool" on Chrome/Brave and opera.
> 3. Export cookies and import into opera
> 4. yt-dlp --cookies-from-browser opera --cookies cookie.txt
> 
> I know this isn't solving the issue but for many it will create workaround.

good idea, sounds practical, I will give it a shot.

**Comment by RichardWinters** [comment]
> Another workaround. Personally I'm fine with just reading cookies from my main Firefox profile, but if desired, you may create a dedicated profile:
> 
> 1. Start Firefox with a new profile, in the specified `<profile dir>`:
> 
> ```
> "C:\Program Files\Mozilla Firefox\firefox.exe" --no-remote --profile "<profile dir>"
> ```
> 
> * The first call of the command will automatically create the profile.
> * The `--no-remote` option lets you run it concurrently with other Firefox profiles.
> 
> 2. On that Firefox instance, log into YouTube, etc. You may then close the Firefox instance.
> 3. Use yt-dlp, reading from these cookies:
> 
> ```
> yt-dlp --cookies-from-browser firefox:"<profile dir>" https://www.youtube.com/watch?v=dQw4w9WgXcQ
> ```

good idea, sounds practical, I will give it a shot.

**Comment by RichardWinters** [comment]
> > Another workaround. Personally I'm fine with just reading cookies from my main Firefox profile, but if desired, you may create a dedicated profile:
> > 
> > 1. Start Firefox with a new profile, in the specified `<profile dir>`:
> > 
> > ```
> > "C:\Program Files\Mozilla Firefox\firefox.exe" --no-remote --profile "<profile dir>"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * The first call of the command will automatically create the profile.
> > * The `--no-remote` option lets you run it concurrently with other Firefox profiles.
> > 
> > 
> > 2. On that Firefox instance, log into YouTube, etc. You may then close the Firefox instance.
> > 3. Use yt-dlp, reading from these cookies:
> > 
> > ```
> > yt-dlp --cookies-from-browser firefox:"<profile dir>" https://www.youtube.com/watch?v=dQw4w9WgXcQ
> > ```
> 
> good idea, sounds practical, I will give it a shot.

@vlakoff Thx, finally it worked using firfox!
However, shortly after successfully using your method, I encountered an error that required a login verification. I suspect it was due to YouTube's anti-scraping mechanism that prevented further downloads. The specific duration of this window is unknown, but it's roughly between 5 to 10 minutes. My internet speed isn't fast, so when downloading a playlist of 46 videos, I only managed to download 5 videos.

`ERROR: [youtube] HkBb1TgOVvU: Sign in to confirm you’re not a bot. Use --cookies-from-browser or --cookies for the authentication. See  https://github.com/yt-dlp/yt-dlp/wiki/FAQ#how-do-i-pass-cookies-to-yt-dlp  for how to manually pass cookies. Also see  https://github.com/yt-dlp/yt-dlp/wiki/Extractors#exporting-youtube-cookies  for tips on effectively exporting YouTube cookies`

**Comment by vlakoff** [comment]
A hypothesis is that YouTube detected it is a fresh session, that has been immediately used for scraping.

* If you open Firefox with the fresh profile and go to YouTube, what happens? Do you encounter any confirmation steps?
* Do you encounter the same issue when using yt-dlp with your regular Firefox profile?

**Comment by bashonly** [comment]
This discussion is veering off-topic. Follow the guide in https://github.com/yt-dlp/yt-dlp/wiki/Extractors#exporting-youtube-cookies , see #8227

**Comment by Butterfly-Dragon** [comment]
I got to this solution inferring what was written here: https://api-minecraft.net/kll9x6ge.png while if one knows what one should do it is "clear eonugh" from what was written there.

It was never spelled out properly and today i spent 4 hours reverse engineering what i did last time to make this work, because it worked on my old machine but not on my new machine.

Currently i am using Version 143.0.7499.193 (Official Build) (64-bit) and it works.

1. Either (a) open start menu start writing "group policies" and it will open "edit group policies" search for "Enable Application Bound Encryption" and DISABLE it or (b) use regedit and go to `HKEY_LOCAL_MACHINE\Software\Policies\Google\Chrome` (you may need to create the "Google" and "Chrome" keys if they are not there) and set a `Dword` named  `ApplicationBoundEncryptionEnabled` to 0
2. install Google Chrome only AFTER that. If you already have Chrome installed (because you needed it to be able to set the policy), you have ***__uninstall__*** it and then reinstall it ***__again__*** because it needs to rebuild every single database and stuff FROM SRATCH (simply deleting the cookies will not work).
4. Log into every site you need cookies from, for every script you have that uses yt-dlp to extract the videos from.
5. Shut down every instance of Chrome, this includs the site apps and the notification tray icon in case you have some background chrome process. Chrome needs to be not open at all for the next step to work.
6. if you did the steps 1-4 `yt-dlp --cookies-from-browser chrome --cookies "cookie-file-random-name.txt"` will extract all your chrome cookies into a single file which you can then use to ...well... do whatever you want with them. Usually to then run yt-dlp with your cookies without having to shut down chrome until the cookies become obsolete again.

NOTE: `cookie-file-random-name.txt` needs to be a file that does not exist for it to be written, if you try and write over a file that does exist it will instead attempt to use that file as a cookies file.

**Comment by bashonly** [comment]
A note about the above workaround:

- You will be disabling a security feature that prevents your cookies from being stolen

- **It will wipe all of your cookies**
